web/grunt/karma-config.js

```diff
@@ -0,0 +1,41 @@
+module.exports = function (grunt) {
+    'use strict';
+
+    const unitTests =  grunt.config('config.tests');
+
+    const karmaTasksConfig = {
+        browsers: ['Firefox'],
+        frameworks: ['jasmine', 'browserify'],
+        preprocessors: {
+            [unitTests]: ['browserify'],
+            '<%= config.compiledScriptsDir %>/**/*.js': ['browserify']
+        },
+        browserify: {
+            debug: true,
+            paths: '<%= config.scriptsDir %>'
+        }
+    };
+
+    const configure = function () {
+        grunt.config.merge({
+            karma: {
+                options: {
+                    files: ['<%= config.compiledScriptsDir %>/**/*.js', unitTests]
+                },
+                dev: karmaTasksConfig,
+                ci: Object.assign(
+                    {
+                        options: {
+                            singleRun: true
+                        }
+                    },
+                    karmaTasksConfig
+                )
+            }
+        });
+    };
+
+    return {
+        configure
```

Comment:
Why return a configure function? I think all the grunt config and task registration should happen immediately, like all other grunt plugins. Then you can `grunt.loadTasks('./grunt')` and write these like grunt plugins.